nixos/modules/services/network-filesystems/ipfs.nix

```diff
@@ -9,7 +9,10 @@ let
 
   ipfsFlags = ''${if cfg.autoMigrate then "--migrate" else ""} ${if cfg.enableGC then "--enable-gc" else ""} ${toString cfg.extraFlags}'';
 
-  pathEnv = { IPFS_PATH = cfg.dataDir; };
+  # Before Version 17.09, ipfs would always use "/var/lib/ipfs/.ipfs" as it's dataDir
+  defaultDataDir = if versionAtLeast config.system.stateVersion "17.09" then
+    "/var/lib/ipfs" else
+    "/var/lib/ipfs/.ipfs";
 
```

Comment:
LGTM

However, that means that users need to rename the IPFS directory (manually) when upgrading.
We could also put a `[[ -d /var/lib/ipfs/.ipfs ]] && mv /var/lib/ipfs/.ipfs/* /var/lib/ipfs && rmdir /var/lib/ipfs/.ipfs` in the
ipfs-init service.
But given that IPFS is not used that much currently in the Nix community (I presume), we should keep that out (this just creates technical debt).